src/components/FormProduct/index.jsx

```diff
@@ -14,88 +14,139 @@ export const FormProduct = () => {
     numberOfPurchases: 0,
     estimatedDaysNextPurchase: null,
   };
-
   const [product, setProduct] = useState(initialStateProduct);
-
-  const [error, setError] = useState('');
-  const handleInputProduct = (e) => {
-    setProduct({
-      ...product,
-      [e.target.name]: e.target.value,
-    });
-    setError(
-      e.target.value ? (
-        ''
-      ) : (
-        <span aria-live="polite">All fields are required</span>
-      ),
-    );
+  const [error, setError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState(false);
+  const [successSave, setSuccessMsg] = useState(false);
+  const handleNameInput = (e) => {
+    const clone = { ...product };
+    clone.item = e.target.value;
+    setProduct(clone);
+    setError(false);
+    setErrorMsg('');
+    setSuccessMsg(false);
   };
+  const handleRadioNextPurchase = (e) => {
+    const clone = { ...product };
+    clone.nextPurchase = e.target.value;
+    setProduct(clone);
+    setError(false);
+    setErrorMsg('');
+    setSuccessMsg(false);
+  };
+
   const handleSubmitProduct = (e) => {
     e.preventDefault();
     const { item, nextPurchase } = product;
 
-    if (isProductDuplicated(products, item))
-      return setError('The item is already on the list');
-
-    const editedProduct = {
-      ...product,
-      nextPurchase: Number(nextPurchase),
-      creationDate: new Date(),
-    };
-    addProduct(editedProduct);
-    setProduct({ ...initialStateProduct });
+    if (item.length > 0) {
+      if (nextPurchase > 0) {
+        if (isProductDuplicated(products, item)) {
+          setError(true);
+          setErrorMsg('The item is already on the list');
+        } else {
+          const editedProduct = {
+            ...product,
+            nextPurchase: Number(nextPurchase),
+            creationDate: new Date(),
+          };
+          addProduct(editedProduct);
+          setProduct({ ...initialStateProduct });
+          setError(false);
+          setErrorMsg('');
+          setSuccessMsg(true);
+        }
+      } else {
+        setError(true);
+        setErrorMsg('Next Purchase is required');
```

Comment:
Next Purchase *date*

Reply:
Thanks for your observation. Fixed!

![image](https://user-images.githubusercontent.com/8689897/109531476-4f575400-7a8e-11eb-889b-1e38a50baee2.png)